### ISSUE TITLE
localhost: Failed to load resource: Request header field token is not allowed by Access-Control-Allow-Headers. 

### ISSUE THREAD

**Issue by cbaksik** [ExLibrisGroup/primo-explore-devenv#123]
I am following the instructions but am unable to use get/post requests from localhost. The console log gives there error:

[Error] XMLHttpRequest cannot load https://...  due to access control checks.

How do I get around this?

**Comment by ajones3066** [comment]
Corinna,

do you have the port number in the proxy URL in the gulp config? the other
thing I have seen is that there is a resource we want to query in an addon,
but our domain has not been allowed by CORS. browser security restrictions
will prohibit the POST if the referring domain is not added.

A

**Comment by cbaksik** [comment]
Yes, I followed the instructions precisely and use a port number. I ended up finding a Chrome extension (I normally use Safari) that will allow cross-origin requests so I installed that, so this isn't urgent, but I do feel like there should be something about this in the doc.

**Comment by noamamit92** [comment]
@cbaksik ,

Can you specify the lime of code that is causing the problem?

If the problem is indeed with cross domain http calls - This is a browser constraint and not related to the development environment at all, it would have happened if you developed this on any other framework.

In that case, I don't think documentation is needed, We do not document general web development methods or best practices.

**Comment by cbaksik** [comment]
Thanks for responding. The problem ended up being that Angular by default, apparently, sends a header token field when sending get requests. The HathiTrust API didn't explicitly permit those, so I was getting the CORS errors, and then I learned I can stop angular from doing that, and then things worked. I also learned that there is a Chrome extension called "Cross Domain" that allows content despite CORS errors, which was very helpful for troubleshooting.  In the end, it turned out to be quite straightforward to omit the token, by setting it to undefined (as I learned on https://docs.angularjs.org/api/ng/service/$http).

return $http({
                    method: 'GET',
                    headers: {
                        'Token': undefined
                      },
                    url: 'https:...',
                    timeout:5000
                })
